docs/experimental/examples/oauth2/globus_app.rst

```diff
@@ -1,5 +1,203 @@
+
+*   :py:class:`~UserApp`, a GlobusApp for interactions between an end user and Globus
+    services. Operations are performed as a *user identity*.
+
+*   :py:class:`~ClientApp`, a GlobusApp for interactions between a client
+    (i.e. service account) and Globus services. Operations are performed as a
+    *client identity*.
+
+Setup
+-----
+
+A GlobusApp is a heavily configurable object. For common scripting usage however,
+instantiation only requires two parameters:
+
+#.  **App Name** - A human readable slug to identify your app in HTTP requests and token
+    caches.
+
+#.  **Client Info** - either a *Native Client's* ID or a *Confidential Client's* ID and
+    secret pair.
+
+    *   There are important distinctions to consider when choosing your client type; see
+        `Developing an Application Using Globus Auth <https://docs.globus.org/api/auth/developer-guide/#developing-apps>`_.
+
+        A simplified heuristic to help choose the client type however is:
+
+        *   Use a *Confidential Client* when your client needs to own cloud resources
+            itself and will be used in a trusted environment where you can securely
+            hold a secret.
+
+        *   Use a *Native Client* when your client will be facilitating interactions
+            between a user and a service, particularly if it is bundled within a
+            script or cli tool to be distributed to end-users' machines.
+
+
+    ..  Note::
+
+        Both UserApps and ClientApps require a client.
+
+        In a UserApp, the client sends requests representing a user identity; whereas in
+        a ClientApp, the requests represent the client identity itself.
+
+
+Once instantiated, a GlobusApp can be passed to any service client using the init
+``app`` keyword argument (e.g. ``TransferClient(app=my_app)``). Doing this will bind the
+app to the client, registering a default set of scopes requirements for the service
+client's resource server and configuring the app as the service client's auth provider.
+
+
+..  tab-set::
+
+    ..  tab-item:: UserApp
+
+        Construct a UserApp then bind it to a Transfer client and a Flows client.
+
+        ..  Note::
+
+            ``UserApp.__init__(...)`` also accepts a `client_secret` keyword argument
+            which must be supplied for confidential clients.
+
+        ..  code-block:: python
+
+            import globus_sdk
+            from globus_sdk.experimental.globus_app import UserApp
+
+            CLIENT_ID = "..."
+            my_app = UserApp("my-user-app", client_id=CLIENT_ID)
+
+            transfer_client = globus_sdk.TransferClient(app=my_app)
+            flows_client = globus_Sdk.FlowsClient(app=my_app)
+
+
+    .. tab-item:: ClientApp
+
+        Construct a ClientApp, then bind it to a Transfer client and a Flows client.
+
+        ..  Note::
+
+            ``ClientApp.__init__(...)`` requires the `client_secret` keyword argument.
+            Native clients, which lack secrets, are not allowed.
```

Comment:
Should this read "Native client IDs, which lack..."?

```suggestion
            Native client IDs, which lack secrets, are not allowed.
```

Reply:
I don't think so, no.

You're not allowed to use native clients in a ClientApp. The client ID is the identifier for that native client that you aren't allowed to use.